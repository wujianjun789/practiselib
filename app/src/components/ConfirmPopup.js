import React from 'react';
import PropTypes from 'prop-types';
import PanelFooter from './PanelFooter';
import {FormattedMessage} from 'react-intl';

const ConfirmPopup = ({tips='', iconClass='', cancel, confirm}) => {
    return <div className="usr-exit">
        <div className='tips'>
            <div><span className={`icon ${iconClass}`}></span></div>
            {tips}
        </div>
         <div className="btn-toolbar">
            <div className='btn-group'><button className="btn btn-default" onClick={cancel}><FormattedMessage id='button.cancel'/></button></div>
            <div className='btn-group'><button className="btn btn-primary" onClick={confirm}><FormattedMessage id='button.confirm'/></button></div>
        </div> 
    </div>
}

ConfirmPopup.prototype = {
    tips: PropTypes.oneOfType([
        PropTypes.string,
        PropTypes.node
    ]).isRequired,
    iconClass: PropTypes.string.isRequired,
    cancel: PropTypes.func.isRequired,
    confirm: PropTypes.func.isRequired
}

export default ConfirmPopup;