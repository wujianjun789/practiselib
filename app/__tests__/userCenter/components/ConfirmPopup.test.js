import React from 'react';
import {shallow, mount} from 'enzyme';
import renderer from 'react-test-renderer';
import ConfirmPopup from '../../../src/components/ConfirmPopup';

describe('<ConfirmPopup />', () => {
    it('default render', () => {
        const click = jest.fn();
        const cmp = shallow(<ConfirmPopup tips='tips' iconClass='icon-class' cancel={click} confirm={click}/>);

        let icon = cmp.find('.icon.icon-class');
        expect(icon.length).toBe(1);

        let txt = cmp.find('.tips');
        expect(txt.text()).toBe('tips');

        let btnCancel = cmp.find('.btn.btn-default');
        expect(btnCancel.text()).toBe('取消');

        let btnConfirm = cmp.find('.btn.btn-primary');
        expect(btnConfirm.text()).toBe('确认');
    });

    it('click simulate', () => {
        const click = jest.fn();
        const cmp = shallow(<ConfirmPopup tips='tips' iconClass='icon-class' cancel={click} confirm={click}/>);

        let btnCancel = cmp.find('.btn.btn-default');
        btnCancel.simulate('click');
        expect(click).toHaveBeenCalledTimes(1);

        let btnConfirm = cmp.find('.btn.btn-primary');
        btnConfirm.simulate('click');
        expect(click).toHaveBeenCalledTimes(2);
    });

    it('snapshot', () => {
        const click = jest.fn();
        const cmp = renderer.create(<ConfirmPopup tips='tips' iconClass='icon-class' cancel={click} confirm={click}/>);
        let tree = cmp.toJSON();
        expect(tree).toMatchSnapshot();
    });
});