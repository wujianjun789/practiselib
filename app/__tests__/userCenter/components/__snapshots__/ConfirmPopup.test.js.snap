exports[`<ConfirmPopup /> snapshot 1`] = `
<div
  className="usr-exit">
  <div
    className="tips">
    <div>
      <span
        className="icon icon-class" />
    </div>
    tips
  </div>
  <div
    className="btn-toolbar">
    <div
      className="btn-group">
      <button
        className="btn btn-default"
        onClick={[Function]}>
        取消
      </button>
    </div>
    <div
      className="btn-group">
      <button
        className="btn btn-primary"
        onClick={[Function]}>
        确认
      </button>
    </div>
  </div>
</div>
`;
